pkg/common/plugin/sshpop/handshake.go

```diff
@@ -137,10 +140,18 @@ func (s *ServerHandshake) VerifyAttestationData(data []byte) error {
 		return Errorf("failed to check host key: %v", err)
 	}
 	s.cert = cert
+	s.hostname = decanonicalizeHostname(cert.ValidPrincipals[0], s.s.canonicalDomain)
 	s.state = stateAttestationDataVerified
 	return nil
 }
 
+func decanonicalizeHostname(fqdn, domain string) string {
+	if domain == "" {
+		return fqdn
+	}
+	return strings.TrimSuffix(fqdn, "."+domain)
```

Comment:
is it worth failing if the fqdn does not have a matching suffix?

Reply:
it couldn't hurt to fail in this case, since it represents a misconfiguration by the client